js/web/lib/wasm/jsep/webgpu/ops/pool.ts

```diff
@@ -51,30 +49,92 @@ const getAdjustedPoolAttributesAndOutputShape = <AttributeType extends AveragePo
   return [newAttributes, isChannelsLast ? outputShapeAsChannelLast : outputShapeAsChannelFirst];
 };
 
-const generatePoolingCode = <AttributeType extends AveragePoolAttributes|MaxPoolAttributes>(
-    shaderHelper: ShaderHelper, x: IndicesHelper, xShape: readonly number[], outputShape: readonly number[],
-    attributes: AttributeType, op1: string, op2: string, start: string): string => {
+const getUniformAndPadInfo = <AttributeType extends AveragePoolAttributes|MaxPoolAttributes>(
+    xShape: readonly number[], outputShape: readonly number[], outputDataType: string, start: number,
+    startType: ProgramUniform['type'],
+    attributes: AttributeType): [ProgramUniform[], UniformsArrayType, boolean, boolean, boolean] => {
   const isChannelsLast = attributes.format === 'NHWC';
   const inputDims = xShape;
-  const dataType = x.type.value;
   const rank = inputDims.length;
   const outputSize = ShapeUtil.size(outputShape);
-  const output = outputVariable('output', x.type.tensor, outputShape);
-
+  const kernelSize = ShapeUtil.size(attributes.kernelShape);
+  const programUniforms: ProgramUniform[] =
+      [{type: 'uint32', data: outputSize}, {type: 'uint32', data: kernelSize}, {type: startType, data: start}];
+  const uniforms: UniformsArrayType = [
+    {name: 'outputSize', type: 'u32'}, {name: 'kernelSize', type: 'u32'},
+    {name: 'start', type: outputDataType as UniformDataElementType}
+  ];
   if (attributes.kernelShape.length <= 2) {
     const kw = attributes.kernelShape[attributes.kernelShape.length - 1];
     const sw = attributes.strides[attributes.strides.length - 1];
     const pwStart = attributes.pads[attributes.pads.length / 2 - 1];
     const pwEnd = attributes.pads[attributes.pads.length - 1];
     const dimIdxW = rank - (isChannelsLast ? 2 : 1);
+    const pwStartEnd = !!(pwStart + pwEnd);
+    programUniforms.push(
+        {type: 'uint32', data: kw},
+        {type: 'uint32', data: sw},
+        {type: 'uint32', data: pwStart},
+        {type: 'uint32', data: pwEnd},
+        {type: 'uint32', data: dimIdxW},
+    );
+    uniforms.push(
+        {name: 'kw', type: 'u32'}, {name: 'sw', type: 'u32'}, {name: 'pwStart', type: 'u32'},
+        {name: 'pwEnd', type: 'u32'}, {name: 'dimIdxW', type: 'u32'});
+
+    let phStartEnd = false;
+    if (attributes.kernelShape.length === 2) {
+      const kh = attributes.kernelShape[attributes.kernelShape.length - 2];
+      const sh = attributes.strides[attributes.strides.length - 2];
+      const phStart = attributes.pads[attributes.pads.length / 2 - 2];
+      const phEnd = attributes.pads[attributes.pads.length - 2];
+      const dimIdxH = rank - (isChannelsLast ? 3 : 2);
+      const dimH = inputDims[dimIdxH];
```

Comment:
Can we directly use `uniforms.x_shape[dimIdxH]` instead of generating a new uniform value?